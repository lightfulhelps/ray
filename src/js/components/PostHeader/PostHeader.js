// @flow
import React from 'react';
import classNames from 'classnames';
import formatDate from 'date-fns/format';

type Props = {
  className?: string,
  date?: Date | number | string,
  dateFormat?: string,
  title: string,
};

const PostHeader = ({ className, date, dateFormat = 'D MMM [at] hh:mm', title }: Props) => {
  const classes = classNames(className, 'post-header', 'd-flex mt-2 mr-2 mb-1 ml-2');
  return (
    <div className={classes}>
      <div>
        <h1 className="h5">{title}</h1>
        <div className="small text-uppercase">{formatDate(date, dateFormat)}</div>
      </div>
    </div>
  );
};
export default PostHeader;
