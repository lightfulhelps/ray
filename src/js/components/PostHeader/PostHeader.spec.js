import React from 'react';
import { shallow } from 'enzyme';
import PostHeader from './PostHeader';

const setup = (overrides = {}) => {
  const props = {
    campaign: {
      name: 'Campaign',
      color: '#ff0000',
    },
    date: '2018-08-22 14:34',
    title: 'Hope For Children',
    ...overrides,
  };
  const wrapper = shallow(<PostHeader {...props} />);

  return {
    wrapper,
    props,
  };
};

describe('<PostHeader />', () => {
  it('should render', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle the className prop', () => {
    const { wrapper } = setup();

    wrapper.setProps({ className: 'custom' });

    expect(wrapper.hasClass('post-header')).toBe(true);
    expect(wrapper.hasClass('custom')).toBe(true);
  });

  it('should optionally display date', () => {
    const { wrapper } = setup();

    expect(wrapper.find('.post-date').exists()).toBe(true);

    wrapper.setProps({ date: null });

    expect(wrapper.find('.post-date').exists()).toBe(false);
  });

  it('should handle the dateFormat prop', () => {
    const { wrapper } = setup();

    expect(wrapper.find('.post-date').text()).toEqual('22 Aug at 14:34');

    wrapper.setProps({ dateFormat: 'HH:MM [on] DD-MM-YYYY' });

    expect(wrapper.find('.post-date').text()).toEqual('14:08 on 22-08-2018');
  });

  it('should handle the campaign prop', () => {
    const { wrapper, props } = setup();

    expect(wrapper.find('.campaign-tag').prop('color')).toEqual(props.campaign.color);
    expect(wrapper.find('.campaign-tag').prop('children')).toEqual(props.campaign.name);
  });

  it('should optionally display the campaign tag', () => {
    const { wrapper } = setup();

    expect(wrapper.find('.campaign-tag').exists()).toBe(true);

    wrapper.setProps({ campaign: null });

    expect(wrapper.find('.campaign-tag').exists()).toBe(false);
  });
});
