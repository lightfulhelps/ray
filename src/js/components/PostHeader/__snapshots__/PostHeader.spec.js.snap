// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostHeader /> should render 1`] = `
<div
  className="post-header d-flex mt-2 mr-2 mb-1 ml-2"
>
  <div
    style={
      Object {
        "height": "35px",
        "width": "35px",
      }
    }
  >
    <Avatar />
  </div>
  <div
    className="mx-1"
  >
    <h1
      className="h5"
    >
      Hope For Children
    </h1>
    <div
      className="post-date small text-uppercase"
    >
      22 Aug at 14:34
    </div>
    <Badge
      className="campaign-tag"
      color="#ff0000"
    >
      Campaign
    </Badge>
  </div>
</div>
`;
